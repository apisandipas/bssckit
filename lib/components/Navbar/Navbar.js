"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _Nav = require("../Nav");

var _theme = _interopRequireDefault(require("../../theme"));

var _themeFunctions = require("../../utils/themeFunctions");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n  position: relative;\n  display: flex;\n  flex-wrap: wrap;\n  align-items: center;\n  justify-content: space-between;\n  padding: ", ";\n  ", "\n  ", "\n  ", "\n  ", "\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n    @media (max-width: ", ") {\n      & > nav:not(:last-child) {\n        display: flex !important;\n        width: 100%;\n        justify-content: space-between;\n      }\n    }\n  "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n    background-color: ", ";\n  "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n    border-color: ", ";\n  "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    color: ", ";\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var color = function color(props) {
  return (0, _styledComponents.css)(_templateObject(), (0, _themeFunctions.getColor)(props, 'navbar', 'color'));
};

var borderColor = function borderColor(props) {
  return (0, _styledComponents.css)(_templateObject2(), (0, _themeFunctions.getColor)(props, 'navbar', 'borderColor'));
};

var backgroundColor = function backgroundColor(props) {
  return (0, _styledComponents.css)(_templateObject3(), (0, _themeFunctions.getColor)(props, 'navbar', 'backgroundColor'));
};

var navbarExpand = function navbarExpand(props) {
  return (props.expandSm || props.expandMd || props.expandLg || props.expandXl) && (0, _styledComponents.css)(_templateObject4(), (0, _themeFunctions.getBreakpointSize)(props));
};

var Navbar = (0, _styledComponents["default"])(_Nav.Nav)(_templateObject5(), function (props) {
  return (0, _themeFunctions.getPadding)(props, 'navbar', 'default');
}, function (props) {
  return color(props);
}, function (props) {
  return borderColor(props);
}, function (props) {
  return backgroundColor(props);
}, function (props) {
  return navbarExpand(props);
});
Navbar.defaultProps = {
  theme: _theme["default"]
};
var _default = Navbar;
exports["default"] = _default;