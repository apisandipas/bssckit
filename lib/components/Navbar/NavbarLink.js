"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _Nav = require("../Nav");

var _theme = _interopRequireDefault(require("../../theme"));

var _themeFunctions = require("../../utils/themeFunctions");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _templateObject12() {
  var data = _taggedTemplateLiteral(["\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n"]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _taggedTemplateLiteral(["\n    white-space: nowrap;\n  "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteral(["\n    line-height: inherit;\n  "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteral(["\n    font-size: ", ";\n  "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteral(["\n    padding-top: ", ";\n    padding-bottom: ", ";\n  "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteral(["\n    display: inline-block;\n  "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n    color: ", ";\n    &:hover,\n    &:focus {\n      color: ", ";\n    }\n  "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n      color: ", ";\n      &:hover,\n      &:focus {\n        color: ", ";\n      }\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n      color: ", ";\n      &:hover,\n      &:focus {\n        color: ", ";\n      }\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n      color: ", ";\n      &:hover,\n      &:focus {\n        color: ", ";\n      }\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n      color: ", ";\n      &:hover,\n      &:focus {\n        color: ", ";\n      }\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n      color: ", ";\n      &:hover,\n      &:focus {\n        color: ", ";\n      }\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var color = function color(props) {
  if (props.dark && props.disabled) {
    return (0, _styledComponents.css)(_templateObject(), (0, _themeFunctions.getColor)(props, 'navbarLink', 'colorDisabled'), (0, _themeFunctions.getColor)(props, 'navbarLink', 'colorDisabledHoverFocus'));
  } else if (props.dark && props.active) {
    return (0, _styledComponents.css)(_templateObject2(), (0, _themeFunctions.getColor)(props, 'navbarLink', 'colorActive'), (0, _themeFunctions.getColor)(props, 'navbarLink', 'colorActiveHoverFocus'));
  } else if (props.dark) {
    return (0, _styledComponents.css)(_templateObject3(), (0, _themeFunctions.getColor)(props, 'navbarLink', 'color'), (0, _themeFunctions.getColor)(props, 'navbarLink', 'colorHoverFocus'));
  } else if (props.disabled) {
    return (0, _styledComponents.css)(_templateObject4(), (0, _themeFunctions.getColor)(props, 'navbarLink', 'colorDisabled'), (0, _themeFunctions.getColor)(props, 'navbarLink', 'colorDisabledHoverFocus'));
  } else if (props.active) {
    return (0, _styledComponents.css)(_templateObject5(), (0, _themeFunctions.getColor)(props, 'navbarLink', 'colorActive'), (0, _themeFunctions.getColor)(props, 'navbarLink', 'colorActiveHoverFocus'));
  }

  return (0, _styledComponents.css)(_templateObject6(), (0, _themeFunctions.getColor)(props, 'navbarLink', 'color'), (0, _themeFunctions.getColor)(props, 'navbarLink', 'colorHoverFocus'));
};

var display = function display(props) {
  return props.brand && (0, _styledComponents.css)(_templateObject7());
};

var padding = function padding(props) {
  return props.brand && (0, _styledComponents.css)(_templateObject8(), (0, _themeFunctions.getPadding)(props, 'navbarLink', 'brandTop'), (0, _themeFunctions.getPadding)(props, 'navbarLink', 'brandBottom'));
};

var fontSize = function fontSize(props) {
  return props.brand && (0, _styledComponents.css)(_templateObject9(), (0, _themeFunctions.getFontSize)(props, 'navbarLink', 'lg'));
};

var lineHeight = function lineHeight(props) {
  return props.brand && (0, _styledComponents.css)(_templateObject10());
};

var whiteSpace = function whiteSpace(props) {
  return props.brand && (0, _styledComponents.css)(_templateObject11());
};

var NavbarLink = (0, _styledComponents["default"])(_Nav.NavLink)(_templateObject12(), function (props) {
  return color(props);
}, function (props) {
  return display(props);
}, function (props) {
  return padding(props);
}, function (props) {
  return fontSize(props);
}, function (props) {
  return lineHeight(props);
}, function (props) {
  return whiteSpace(props);
});
NavbarLink.defaultProps = {
  theme: _theme["default"]
};
var _default = NavbarLink;
exports["default"] = _default;