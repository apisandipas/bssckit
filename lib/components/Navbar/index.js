"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Navbar", {
  enumerable: true,
  get: function get() {
    return _Navbar["default"];
  }
});
Object.defineProperty(exports, "NavbarLink", {
  enumerable: true,
  get: function get() {
    return _NavbarLink["default"];
  }
});

var _Navbar = _interopRequireDefault(require("./Navbar"));

var _NavbarLink = _interopRequireDefault(require("./NavbarLink"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }