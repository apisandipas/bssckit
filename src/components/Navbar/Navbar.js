import styled, { css } from 'styled-components';
import { Nav } from '../Nav';
import defaultTheme from '../../theme';
import { getColor, getPadding, getBreakpointSize } from '../../utils/themeFunctions';

const color = props =>
  css`
    color: ${getColor(props, 'navbar', 'color')};
  `;

const borderColor = props =>
  css`
    border-color: ${getColor(props, 'navbar', 'borderColor')};
  `;

const backgroundColor = props =>
  css`
    background-color: ${getColor(props, 'navbar', 'backgroundColor')};
  `;

const navbarExpand = props =>
  (props.expandSm || props.expandMd || props.expandLg || props.expandXl) &&
  css`
    @media (max-width: ${getBreakpointSize(props)}) {
      & > nav:not(:last-child) {
        display: flex !important;
        width: 100%;
        justify-content: space-between;
      }
    }
  `;

const Navbar = styled(Nav)`
  position: relative;
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-between;
  padding: ${props => getPadding(props, 'navbar', 'default')};
  ${props => color(props)}
  ${props => borderColor(props)}
  ${props => backgroundColor(props)}
  ${props => navbarExpand(props)}
`;

Navbar.defaultProps = {
  theme: defaultTheme,
};

export default Navbar;
