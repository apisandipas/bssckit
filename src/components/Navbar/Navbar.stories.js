import React, { useState } from 'react';
import { Story } from '../../utils/functions';
import { Navbar, NavbarLink } from '../Navbar';
import { Nav } from '../Nav';
import { Button } from '../Button';

export default {
  title: 'Navbar',
};

export const dark = Story(() => {
  const [hidden, setHidden] = useState(true);

  return (
    <Navbar expandSm dark>
      <Nav start>
        <NavbarLink dark brand href='#'>
          BRAND
        </NavbarLink>
        <Nav end>
          <Button dark outline toggleCollapse expandSm onClick={() => setHidden(!hidden)}>
            <span>&#9776;</span>
          </Button>
        </Nav>
      </Nav>
      <Nav start collapse expandSm hidden={hidden}>
        <NavbarLink dark active>
          Active
        </NavbarLink>
        <NavbarLink dark href='#'>
          Link
        </NavbarLink>
        <NavbarLink dark href='#'>
          Link
        </NavbarLink>
        <NavbarLink dark disabled href='#'>
          Disabled
        </NavbarLink>
      </Nav>
    </Navbar>
  );
});

export const light = Story(() => {
  const [hidden, setHidden] = useState(true);

  return (
    <Navbar expandSm light>
      <Nav start>
        <NavbarLink light brand href='#'>
          BRAND
        </NavbarLink>
        <Nav end>
          <Button light outline toggleCollapse expandSm onClick={() => setHidden(!hidden)}>
            <span>&#9776;</span>
          </Button>
        </Nav>
      </Nav>
      <Nav start collapse expandSm hidden={hidden}>
        <NavbarLink light active>
          Active
        </NavbarLink>
        <NavbarLink light href='#'>
          Link
        </NavbarLink>
        <NavbarLink light href='#'>
          Link
        </NavbarLink>
        <NavbarLink light disabled href='#'>
          Disabled
        </NavbarLink>
      </Nav>
    </Navbar>
  );
});
