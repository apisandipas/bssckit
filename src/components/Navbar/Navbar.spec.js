import React from 'react';
import 'jest-styled-components';
import { render } from '@testing-library/react';

import Navbar from './Navbar';

describe('Navbar Component', () => {
  it('matches Navbar', () => {
    const { container } = render(<Navbar />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('position', 'relative');
    expect(container.firstChild).toHaveStyleRule('display', 'flex');
    expect(container.firstChild).toHaveStyleRule('flex-wrap', 'wrap');
    expect(container.firstChild).toHaveStyleRule('align-items', 'center');
    expect(container.firstChild).toHaveStyleRule('justify-content', 'space-between');
    expect(container.firstChild).toHaveStyleRule('padding', '0.5rem 1rem');
  });

  it('matches Navbar dark', () => {
    const { container } = render(<Navbar dark />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('color', 'rgba(255,255,255,0.5)');
    expect(container.firstChild).toHaveStyleRule('border-color', 'rgba(255,255,255,0.1)');
    expect(container.firstChild).toHaveStyleRule('background-color', '#343a40');
  });

  it('matches Navbar light', () => {
    const { container } = render(<Navbar light />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
