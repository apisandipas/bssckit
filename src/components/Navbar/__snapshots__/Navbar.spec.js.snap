// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component matches Navbar 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-left: 0;
  margin-bottom: 0;
  list-style: none;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.5rem 1rem;
  color: rgba(0,0,0,0.5);
  border-color: rgba(0,0,0,0.1);
  background-color: #f0f0f0;
}

.c0 > a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:last-child {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<nav
  class="sc-fznKkj sc-fznJRM c0"
/>
`;

exports[`Navbar Component matches Navbar dark 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-left: 0;
  margin-bottom: 0;
  list-style: none;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.5rem 1rem;
  color: rgba(255,255,255,0.5);
  border-color: rgba(255,255,255,0.1);
  background-color: #343a40;
}

.c0 > a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:last-child {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<nav
  class="sc-fznKkj sc-fznJRM c0"
/>
`;

exports[`Navbar Component matches Navbar light 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-left: 0;
  margin-bottom: 0;
  list-style: none;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.5rem 1rem;
  color: rgba(0,0,0,0.5);
  border-color: rgba(0,0,0,0.1);
  background-color: #f0f0f0;
}

.c0 > a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:last-child {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<nav
  class="sc-fznKkj sc-fznJRM c0"
/>
`;
